test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -131,8 +131,8 @@ public Http2ConnectionTests()
         {
             var inlineSchedulingPipeOptions = new PipeOptions(
                 pool: _memoryPool,
-                readerScheduler: PipeScheduler.Inline,
-                writerScheduler: PipeScheduler.Inline,
+                readerScheduler: PipeScheduler.ThreadPool,
+                writerScheduler: PipeScheduler.ThreadPool,
```

Comment:
This is probably responsible for the ConnectionResetLoggedWithActiveStreams failure. I don't think you should change this. If for some reason it's needed for a new test, use `threadPoolSchedulingPipeOptions` for just the new test.